services/ux-backend/handlers/rotatekeys/handler.go

```diff
@@ -0,0 +1,89 @@
+package rotatekeys
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	contentTypeTextPlain                     = "text/plain"
+	onboardingValidationPublicKeySecretName  = "onboarding-ticket-key"
+	onboardingValidationPrivateKeySecretName = "onboarding-private-key"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, cl client.Client, namespace string) {
+	switch r.Method {
+	case "POST":
+		handlePost(r.Context(), w, cl, namespace)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(ctx context.Context, w http.ResponseWriter, cl client.Client, namespace string) {
+
+	w.Header().Set("Content-Type", contentTypeTextPlain)
+
+	privateKeySecret := &corev1.Secret{}
+	privateKeySecret.Name = onboardingValidationPrivateKeySecretName
+	privateKeySecret.Namespace = namespace
+	err := cl.Get(ctx, client.ObjectKeyFromObject(privateKeySecret), privateKeySecret)
```

Comment:
Why aren't we directly deleting the secret option instead of retrieving it? I assume this secret is only present when odf is installed in provider mode. Correct me if I am wrong

Reply:
well, I thought private key will not exist if public key is being managed by user and so this endpoint shouldn't be called.

as per https://github.com/red-hat-storage/ocs-operator/pull/2469#discussion_r1495874806 calling this ep seems to be opting into keys to be managed by odf.

Reply:
Okay, I understand now. I apologize for missing that comment. So, are you going to remove this check

Reply:
np, done.